### DISCUSSION TITLE
How does the import work?

### DISCUSSION THREAD

**Discussion by logkirk** [afterxleep/Bouncer#81] [Q&A] [answered]
I exported some rules I had created, and the rule structure appears to be in the following format:

```
[
	{
		"action": "junk",
		"id": "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX",
		"type": "any",
		"caseSensitive": false,
		"subAction": "none",
		"phrase": "foobar",
		"useRegex": false
	}
]
```

Is there a reference for the values these fields can take? What is the "id" field and how should this be set if I want to create new rules in JSON format and import them?

**Comment by afterxleep** [comment, marked as answer]
@logkirk the ID is a unique field to identify the filter in the local database.  You can use a [UUID generator](https://www.uuidgenerator.net) for that.

Here's the JSON with all the possible options for the remainder of the fields.    Hope that helps.

```
[
	{
		"action": none | allow | junk | transaction | promotion,
		"id": "Uniquely generated UUID",
		"type": any | sender | message,
		"caseSensitive": true | false,
		"subAction": none | transactionOrder | transactionFinance | transactionReminders | transactionHealth | transactionOther | promotionOffers | promotionCoupons | promotionOther,
		"phrase": "foobar",
		"useRegex": true | false
	}
]
```
